V3

besendorf opened the pull request.

DonnchaC committed:
Run bugreport and backup modules during check-androidqf

Adding support to automatically run ADB backup and bugreport modules
automatically when running the check-androidqf command. This is a first
step to deduplicate the code for Android modules.

DonnchaC committed:
Deduplicate modules which are run by the sub-commands.

DonnchaC committed:
Raise the proper NoAndroidQFBackup exception when a back-up isn't found

DonnchaC committed:
Remove check-adb command and update docs

DonnchaC committed:
Remove check-apk code and old dependencies

DonnchaC committed:
Major refactor to add structured alerting and typed indicators

This commit makes a structural change to MVT by changing binary
detected/not detected logic into a structured multi-level system
of alerts. This gives far more power to extend MVT and manage
alerts.

This commit also begins the process of adding proper typing for
key objects used in MVT including Indicators, IndicatorMatches,
and ModuleResults. This will also be keep to programmatically using
the output of MVT.

DonnchaC committed:
Fix up, remove ADB module base

DonnchaC committed:
Rework old detections tracking into stuctured alert levels

DonnchaC committed:
Quote STIX path in log line

DonnchaC committed:
Fix profile events log line

DonnchaC committed:
Close open archive (zip/tar) file handles

besendorf committed:
Merge refactor/structured-alerting into v2

Resolved conflicts:
- pyproject.toml: Used v2 pinned dependency versions
- Removed cmd_check_adb.py (deleted in refactor branch)
- Updated all command files to include disable_version_check and disable_indicator_check flags
- Adopted new AlertStore system from refactor branch
- Updated version to 3.0.0
- Kept VirusTotal functionality commented out
- Consolidated imports and module lists
- Adopted refactor branch's simplified JSON loading
- Updated iOS modules to use new alertstore approach

besendorf committed:
Fix root_binaries and mounts modules to use alertstore

besendorf committed:
Update tests to use alertstore instead of detected attribute

besendorf committed:
Fix alertstore method calls - use high() instead of warning()

besendorf committed:
Fix remaining test errors

- Add log_latest() call in root_binaries to log each alert
- Fix UnboundLocalError in cmd_check_androidqf by initializing bugreport variable
- Remove incorrect backup.close() call since load_backup() returns bytes
- Remove duplicate from_ab method in cmd_check_backup that was using old attributes

besendorf committed:
Log alerts on add

besendorf committed:
Remove slug from alertstore calls

besendorf committed:
update alerts.py

besendorf committed:
update alerts.py

besendorf committed:
move indicator_match to alert object

besendorf committed:
.

besendorf committed:
- Remove timeline_detected and route to alertstore

besendorf committed:
fix typing for mypy

besendorf committed:
Merge branch 'main' into v2

besendorf committed:
Remove unused type imports

besendorf committed:
Merge branch 'main' into v2 - resolve conflicts in dumpsys_battery_daily and bugreport/base

besendorf committed:
Merge branch 'main' into v2 - resolve formatting conflict in indicators.py

DonnchaC committed:
Fix check_receiver_prefix and check_android_property_name

- check_receiver_prefix() used dict syntax (ioc["value"]) on Indicator
  dataclass objects from get_iocs(). Changed to ioc.value/ioc.name.
- check_receiver_prefix() returned raw ioc instead of IndicatorMatch.
  Now returns IndicatorMatch with descriptive message.
- Fixed return type annotations on both methods to Optional[IndicatorMatch].
- Removed unused Union import.

DonnchaC committed:
Fix residual self.detected usage in packages and dumpsys_receivers

These modules still used self.detected.append() which no longer exists
after the alertstore migration. Converted to alertstore calls:
- packages.py: ROOT_PACKAGES detection → alertstore.high()
- dumpsys_receivers.py: receiver IOC match → alertstore.critical()

DonnchaC committed:
Fix SMS module alertstore.high() call passing slug as message

The first argument was self.get_slug() (module slug) instead of a
human-readable message. The module is already auto-detected via
AlertStore._get_calling_module(). Also removed redundant log_latest().

tesitura commented on src/mvt/common/module.py:
```diff
@@ -116,13 +124,15 @@ def save_to_json(self) -> None:
             with open(detected_json_path, "w", encoding="utf-8") as handle:
-                json.dump(self.detected, handle, indent=4, cls=CustomJSONEncoder)
+                json.dump(
+                    self.alertstore.alerts, handle, indent=4, cls=CustomJSONEncoder
```
```suggestion
                    self.alertstore.as_json(), handle, indent=4, cls=CustomJSONEncoder
```

tesitura commented on src/mvt/ios/modules/net_base.py:
```diff
@@ -232,7 +237,10 @@ def _find_suspicious_processes(self):
 
-                    self.log.warning(msg)
+                    self.alertstore.medium(
+                        self.get_slug(), msg, proc["live_isodate"], proc
```
```suggestion
                        msg, proc["live_isodate"], proc
```

tesitura commented on src/mvt/ios/modules/net_base.py:
```diff
@@ -251,16 +259,23 @@ def check_manipulated(self):
-                    "Found manipulated process entry %s. Entry on %s",
-                    result["live_proc_id"],
+                self.alertstore.high(
+                    self.get_slug(),
```
```suggestion
```

tesitura commented on src/mvt/ios/modules/net_base.py:
```diff
@@ -251,16 +259,23 @@ def check_manipulated(self):
             result["proc_name"] = "MANIPULATED [process record deleted]"
-            self.detected.append(result)
+            self.alertstore.high(
+                self.get_slug(),
```
```suggestion
```

tesitura commented on src/mvt/ios/modules/net_base.py:
```diff
@@ -278,12 +293,13 @@ def find_deleted(self):
-                    'Missing process %d. Previous process at "%s" (%s)',
-                    proc_id,
+                self.alertstore.low(
+                    self.get_slug(),
```
```suggestion
```

tesitura commented on src/mvt/ios/modules/net_base.py:
```diff
@@ -333,7 +349,9 @@ def check_indicators(self) -> None:
+            if ioc_match:
+                result["matched_indicator"] = ioc_match.ioc
+                self.alertstore.critical(
+                    self.get_slug(), ioc_match.message, result["first_isodate"], result
```
```suggestion
                    ioc_match.message, result["first_isodate"], result
```

tesitura commented on src/mvt/ios/modules/fs/filesystem.py:
```diff
@@ -51,19 +56,19 @@ def check_indicators(self) -> None:
-                self.detected.append(result)
+            ioc_match = self.indicators.check_file_path(result["path"])
+            if ioc_match:
+                self.alertstore.high(self.get_slug(), ioc_match.message, "", result)
```
```suggestion
                self.alertstore.high(ioc_match.message, "", result)
```

tesitura commented on src/mvt/ios/modules/fs/filesystem.py:
```diff
@@ -51,19 +56,19 @@ def check_indicators(self) -> None:
-                self.detected.append(result)
+            ioc_match = self.indicators.check_file_path_process(result["path"])
+            if ioc_match:
+                self.alertstore.high(self.get_slug(), ioc_match.message, "", result)
```
```suggestion
                self.alertstore.high(ioc_match.message, "", result)
```

tesitura commented on src/mvt/ios/modules/mixed/safari_history.py:
```diff
@@ -95,9 +100,11 @@ def _find_injections(self):
-                        "Redirect took less than a second! (%d milliseconds)",
-                        elapsed_ms,
+                    self.alertstore.medium(
+                        self.get_slug(),
```
```suggestion
```

tesitura commented on src/mvt/ios/modules/mixed/sms_attachments.py:
```diff
@@ -56,27 +61,30 @@ def serialize(self, record: dict) -> Union[dict, list]:
-                    "Suspicious iMessage attachment %s on %s",
-                    attachment["filename"],
+                self.alertstore.medium(
+                    self.get_slug(),
```
```suggestion
```

tesitura committed:
Apply suggestions from code review

Fix JSON serialization in `module.save_to_json` and fix argument order in iOS alertstore calls.

Co-authored-by: tes <tesitura@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'origin/main' into v2

# Conflicts:
#	src/mvt/android/artifacts/tombstone_crashes.py
#	src/mvt/common/command.py
#	src/mvt/ios/modules/backup/manifest.py

Commit:
Remove unsupported ADB modules

Commit:
Fail removed check-adb command

Commit:
Fix alert serialization and logging

Commit:
Close sqlite connections in iOS modules

tesitura committed:
Fix DEBUG messages not reaching handlers, save_to_json for dictionary results and TypeError on mixed event_time types in safary_history

Commit:
add matched_indicator via alertstore instead of directly modifying json objects

Commit:
Alert on battery daily uninstall and downgrade

besendorf committed:
Lower alert severity to medium for suspicious items

Commit:
Switch version to 2026.4.28 CalVer

besendorf committed:
Merge branch 'main' into v2

hcpspp5q9y-netizen reviewed (commented):
H
